flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/NettyShuffleMaster.java

```diff
@@ -71,7 +75,10 @@ public NettyShuffleMaster(Configuration conf) {
                 conf.getInteger(NettyShuffleEnvironmentOptions.NETWORK_SORT_SHUFFLE_MIN_BUFFERS);
         networkBufferSize = ConfigurationParserUtils.getPageSize(conf);
 
-        if (conf.getBoolean(NETWORK_HYBRID_SHUFFLE_ENABLE_NEW_MODE)) {
+        if ((conf.get(BATCH_SHUFFLE_MODE) == ALL_EXCHANGES_HYBRID_FULL
+                        || conf.get(BATCH_SHUFFLE_MODE) == ALL_EXCHANGES_HYBRID_SELECTIVE)
+                && conf.getBoolean(NETWORK_HYBRID_SHUFFLE_ENABLE_NEW_MODE)
+                && conf.getString(NETWORK_HYBRID_SHUFFLE_REMOTE_STORAGE_BASE_PATH) != null) {
             tieredInternalShuffleMaster = new TieredInternalShuffleMaster(conf);
```

Comment:
In the java doc of `TieredInternalShuffleMaster`, we can see that `All the tiered storage operations with the shuffle master should be wrapped in this class.` I wonder why this is only required when `conf.getString(NETWORK_HYBRID_SHUFFLE_REMOTE_STORAGE_BASE_PATH) != null`?